feat: Shopping list UI overhaul - add wakelock

Wetzel402 opened the pull request:
<!--
  This template provides some ideas of things to include in your PR description.

  To start, try providing a short summary of your changes in the Title above. We follow Conventional Commits syntax, please ensure your title is prefixed with one of:
  - `feat: `
  - `fix: `
  - `docs: `
  - `chore: `

  If a section of the PR template does not apply to this PR, then delete that section.

  PLEASE READ:
  -------------------------
  Mealie is moving to a regular, automatic release schedule. This means that all PRs should be in a
  stable state, ready for release. This includes:

  - Ensuring new tests have been added to cover new features, or to prevent regressions.
  - Work is fully complete and usable

 -->

## What type of PR is this?
<!--
  Delete any of the following that do not apply:
 -->
- feature

## What this PR does / why we need it:
<!--
  What goal is this change working towards?
  Provide a bullet pointed summary of how each file was changed.
  Briefly explain any decisions you made with respect to the changes.
  Include anything here that you didn't include in *Release Notes*
  above, such as changes to CI or changes to internal methods.

  If there is a UI component to the change, please include before/after images.
-->
- Add keep screen awake 
## Which issue(s) this PR fixes:

<!--
If this PR fixes one of more issues, list them here.
One per line, like so:
Fixes #123
Fixes #39
-->
https://github.com/mealie-recipes/mealie/pull/4077
## Testing
<!--
  Describe how you tested this change.
-->
Tested in dev container

Wetzel402 committed:
modified:   frontend/pages/shopping-lists/_id.vue

Kuchenpirat committed:
Merge branch 'mealie-next' into wakelock

Kuchenpirat reviewed (commented):
Thanks for spliting all this up. Makes it super easy to review 😊👍
Since this is all the same code as on the recipe page, could you create a component that includes the switch as well as the functionality and just throw this component on both pages? 
Decreases duplicate code and makes the code of those pages a bit less convoluted.

Wetzel402 commented:
> Thanks for spliting all this up. Makes it super easy to review 😊👍 Since this is all the same code as on the recipe page, could you create a component that includes the switch as well as the functionality and just throw this component on both pages? Decreases duplicate code and makes the code of those pages a bit less convoluted

That makes sense.  It might take me a bit to figure that out since I'm new to `vue`.

Kuchenpirat commented:
No worries. If you want to tackle this for fun all good and take your time, but if this is just work for you i am happy to throw this together in 5 minutes and add this to this PR.

Wetzel402 committed:
modified:   frontend/components/Domain/Recipe/RecipePage/RecipePage.vue
	new file:   frontend/components/global/Wakelock.vue
	modified:   frontend/pages/shopping-lists/_id.vue

Wetzel402 commented:
> No worries. If you want to tackle this for fun all good and take your time, but if this is just work for you i am happy to throw this together in 5 minutes and add this to this PR.

Maybe I'm a masochist but I enjoy putting myself through learning new ways to code...haha

I've pushed the changes for your review.

michael-genson commented on frontend/components/Domain/Recipe/RecipePage/RecipePage.vue:
```diff
@@ -104,6 +98,7 @@ import RecipePageTitleContent from "./RecipePageParts/RecipePageTitleContent.vue
 import RecipePageComments from "./RecipePageParts/RecipePageComments.vue";
 import { useLoggedInState } from "~/composables/use-logged-in-state";
 import RecipePrintContainer from "~/components/Domain/Recipe/RecipePrintContainer.vue";
+import Wakelock from "~/components/global/Wakelock.vue";
```
No need to import since you added it as a global component

michael-genson commented on frontend/components/global/Wakelock.vue:
```diff
@@ -0,0 +1,48 @@
+<template>
```
You can name it something like `WakelockSwitch`

michael-genson reviewed (changes requested):
Check the files changed tab on GH; a few linter warnings got raised. I also added a few comments

michael-genson commented on frontend/components/global/Wakelock.vue:
```diff
@@ -0,0 +1,48 @@
+import { useWakeLock } from "@vueuse/core";
+
+export default defineComponent({
+    setup(props, context) {
```
You can leave this empty if you're not using props/context

```suggestion
    setup() {
```

Wetzel402 committed:
modified:   frontend/components/Domain/Recipe/RecipePage/RecipePage.vue
	renamed:    frontend/components/global/Wakelock.vue -> frontend/components/global/WakelockSwitch.vue
	modified:   frontend/pages/shopping-lists/_id.vue

Kuchenpirat committed:
Merge branch 'mealie-next' into wakelock

Kuchenpirat reviewed (dismissed):
Looks good to me 🚀
Thanks for your work on this!

Ah right, i missed the code warnings.

michael-genson reviewed (changes requested):
There's still a few warnings to take care of; check out the "Files changed" tab on GH and scroll down and you'll see them

Wetzel402 committed:
modified:   frontend/components/global/WakelockSwitch.vue

Wetzel402 commented:
> There's still a few warnings to take care of; check out the "Files changed" tab on GH and scroll down and you'll see them

Sorry for the delay.  PR should be ready now.

Kuchenpirat committed:
Merge branch 'mealie-next' into wakelock

Kuchenpirat commented:
@Wetzel402 any reason you closed this?

Wetzel402 commented:
> @Wetzel402 any reason you closed this?

Sorry, I thought it had been merged already and was doing some pruning on my repository.

Kuchenpirat committed:
Merge branch 'mealie-next' into wakelock

Kuchenpirat reviewed (approved):
Looks all good 👍

Kuchenpirat committed:
Merge branch 'mealie-next' into wakelock

michael-genson committed:
Merge branch 'mealie-next' into wakelock

boc-the-git commented:
Thanks for this @Wetzel402; it was useful to me today! :)
